src/client/metric.rs

```diff
@@ -281,14 +287,178 @@ impl<T: MetricType + Clone> Metric<T> {
     pub fn indom(&self) -> u32 { self.indom }
     pub fn shorthelp(&self) -> &str { &self.shorthelp }
     pub fn longhelp(&self) -> &str { &self.longhelp }
+}
+
+#[derive(Clone)]
+/// An instance domain is a set of instances
+pub struct Indom {
+    pub (super) instances: HashSet<String>,
+    pub (super) id: u32,
+    shorthelp: String,
+    longhelp: String
+}
 
-    pub (super) fn set_mmap_view(&mut self, mmap_view: MmapViewSync) {
-        self.mmap_view = mmap_view;
+impl Indom {
+    /// Creates a new instance domain with given instances, and short and long help text
+    pub fn new(instances: &[&str], shorthelp_text: &str, longhelp_text: &str) -> Self {
+        let mut hasher = DefaultHasher::new();
+        instances.hash(&mut hasher);
+
+        Indom {
+            instances: instances.into_iter().map(|inst| inst.to_string()).collect(),
+            id: (hasher.finish() as u32) & ((1 << INDOM_BIT_LEN) - 1),
```

Comment:
I think this is okay for now but in the future we'll want a way for the user to provide their own name to ID resolver (kind of like `io.pcp.parfait.dxm.IdentifierSource` is used in parfait)